AsyncDisplayKit/Details/ASThread.h

```diff
@@ -29,6 +29,7 @@ static inline BOOL ASDisplayNodeThreadIsMain()
 #ifdef __cplusplus
 
 #define TIME_LOCKER 0
+#define CHECK_LOCKING_SAFETY DEBUG
```

Comment:
This might be a bit too aggressive because lots of warnings will be printed due to our current way of building logging context (PIContextLogger.m). Let me know if you think we should disable this by default.